app/services/publisher_statement_getter.rb

```diff
@@ -75,19 +75,13 @@ def replace_account_identifiers_with_titles(transactions)
   def get_uphold_transactions
     uphold = []
 
+
     publisher.uphold_connection&.uphold_connection_for_channels&.each do |card_connection|
-      transactions = card_connection.uphold_connection.uphold_client.transaction.all(id: card_connection.card_id)
-      next if transactions.blank?
-
-      transactions.each do |transaction|
-        uphold << Statement.new(
-          channel: card_connection.channel.details.publication_title,
-          transaction_type: Statement::UPHOLD_CONTRIBUTION,
-          amount: transaction.origin.dig("amount")&.to_d,
-          settlement_currency: transaction.destination.dig("currency"),
-          settlement_amount: transaction.destination.dig("amount")&.to_d,
-          created_at: transaction.createdAt.to_date,
-        )
+      # Refresh the cache, should only request the most recent page
+      CacheUpholdTips.perform_now(uphold_connection_for_channel_id: card_connection.id)
```

Comment:
I’d keep an eye on this: Might be worthwhile to run this asynchronously if this function takes too long

Reply:
Agreed, I originally separated it out into a job because I thought it might make sense to preload upon login or something similar but am going to keep it here for now.